File: scripts/test_stress_insmod_rmmod.pl
#!/usr/bin/perl -w

# this is a stress testing script which runs many insmod/rmmod combinations
# to check that there are no problems in loading/unloading the driver...
#					Mark Veltzer

use strict;
use diagnostics;

my($module)=$ARGV[0];
my($count)=$ARGV[1];
print "module is [$module]\n";
print "count is [$count]\n";

sub my_system($) {
	my($cmd)=$_[0];
	print "doing [$cmd]\n";
	my($res)=system($cmd);
	if($res!=0) {
		die("failed to do cmd [$cmd] with result [$res]");
	}
	#print "res is [$res]\n";
}

for(my($i)=0;$i<$count;$i++) {
	my_system("sudo insmod $module");
	my_system("sudo rmmod $module");
}

File: scripts/test_stress_insmod_rmmod.py
#!/usr/bin/env python

"""
this is a stress testing script which runs many insmod/rmmod combinations
to check that there are no problems in loading/unloading the driver...
"""

import sys
import os

def main():
    """ main entry point """
    if len(sys.argv)!=3:
        print(f"{sys.argv[0]}: usage: module count")
        sys.exit(1)
    module=sys.argv[1]
    count=sys.argv[2]
    print(f"module is {module}")
    print(f"count is {count}")
    for _ in range(count):
        os.system(f"sudo insmod {module}")
        os.system(f"sudo rmmod {module}")


if __name__ == "__main__":
    main()
